ProcessManager.py

```diff
@@ -42,11 +57,20 @@ def startAllProcesses(self):
         TODO: create a function that starts all processes or threads in "self.T" all at once.
         """
 
+        for process in process_list:
+            process_list[process].set = process_list[process].cmd
```

Comment:
I think this is where the `Popen` constructor should be called.

Reply:
Another thing you could do if you wanted is create another method in `ProcessManager` to start just a single process, and have `startAllProcesses` call that new method for every process.